### PR TITLE
More fixes

### PR DESCRIPTION
**Pull request by rpavlik**
A handful of fixes found when building with CMake. The largest one is a simplification of the source structure - some plugins needed to "link against" meshlab (and not just common) to get ml_selection_buffers, which is a complicated thing to do. Since it was a fairly simple class I just moved it over to common direclty.

### CODE REVIEW

**Comment by rpavlik** [comment]
Updated with a few more small fixes, including a ton of spelling/typo fixes thanks to the "codespell" tool

File: src/meshlabplugins/filter_camera/filter_camera.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ void FilterCameraPlugin::initParameterSet(QAction *action, MeshDocument &/*m*/,
             rotCenter.push_back("camera viewpoint");
             rotCenter.push_back("custom point");
             parlst.addParam(new RichEnum("rotCenter", 0, rotCenter, tr("Center of rotation:"), tr("Choose a method")));
-            parlst.addParam(new RichDynamicFloat("angle",0,-360,360,"Rotation Angle","Angle of rotation (in <b>degree</b>). If snapping is enable this vaule is rounded according to the snap value"));
+            parlst.addParam(new RichDynamicFloat("angle",0,-360,360,"Rotation Angle","Angle of rotation (in <b>degree</b>). If snapping is enable this value is rounded according to the snap value"));
```
Comment (jmespadero) [inline src/meshlabplugins/filter_camera/filter_camera.cpp]: If snapping is **enabled, ** this value is rounded according to the snap value

File: src/meshlabplugins/filter_clean/cleanfilter.cpp [hunk 2]
```diff
@@ -225,7 +225,7 @@ void CleanFilter::initParameterSet(QAction *action,MeshDocument &md, RichParamet
...
           parlst.addParam(new RichFloat("Clustering",20.0f,"Clustering radius (% of ball radius)","To avoid the creation of too small triangles, if a vertex is found too close to a previous one, it is clustered/merged with it."));
           parlst.addParam(new RichFloat("CreaseThr", 90.0f,"Angle Threshold (degrees)","If we encounter a crease angle that is too large we should stop the ball rolling"));
-          parlst.addParam(new RichBool("DeleteFaces",false,"Delete intial set of faces","if true all the initial faces of the mesh are deleted and the whole surface is rebuilt from scratch, other wise the current faces are used as a starting point. Useful if you run multiple times the algorithm with an incrasing ball radius."));
+          parlst.addParam(new RichBool("DeleteFaces",false,"Delete initial set of faces","if true all the initial faces of the mesh are deleted and the whole surface is rebuilt from scratch, other wise the current faces are used as a starting point. Useful if you run multiple times the algorithm with an incrasing ball radius."));
```
Comment (jmespadero) [inline src/meshlabplugins/filter_clean/cleanfilter.cpp]: ...from **scratch. Otherwise** ... the algorithm with **an increasing** ball radius

**Comment by alemuntoni** [comment]
This is huge, thank you @rpavlik!!
Can I ask you to remove the changes from ubsupported plugins? 
Thanks!

File: src/meshlabplugins/edit_align/align/AlignPair.cpp [hunk 3]
```diff
@@ -271,7 +271,7 @@ we have two meshes:
 - Fix the mesh that does not move and stays in the spatial indexing structure.
 - Mov the mesh we 'move' e.g. the one for which we search the transforamtion.
 
-requires normalize normals for vertexes AND faces
+requires normalize normals for vertices AND faces
```
[the remark this answers is not in the document]
Comment (rpavlik) [reply src/meshlabplugins/edit_align/align/AlignPair.cpp]: Ah, interesting, I hadn't heard that, but I did see vertices in a number of places near "vertexes" so I figured I'd let codespell do its thing. Easy enough to drop these.

File: src/meshlabplugins/edit_align/align/Guess.cpp [hunk 4]
```diff
@@ -194,7 +194,7 @@ void Guess::ComputeAlign(std::vector<Result> &rv)
                                 ar.Pmov, ar.Nmov,
                                 ar.H, ar.as);
             ar.err=ar.as.LastPcl50();
-//			printf("Ornt %2i n %3i (%6.3f %6.3f %6.3f) ang %2i %s Err:%8.5f %4i  \n",
+//			printf("Ornt %2i n %3i (%6.3f %6.3f %6.3f) and %2i %s Err:%8.5f %4i  \n",
```
[the remark this answers is not in the document]
Comment (rpavlik) [reply src/meshlabplugins/edit_align/align/Guess.cpp]: ah, thanks for the careful review.

**Comment by rpavlik** [comment]
OK, I have removed changes from the unsupported plugins (maybe they should just be removed from the tree so they don't add to apparent maintenance load? e.g. think they're the only ones that use bzip2 which is getting carried around in external, etc.)

I also removed all changes of "vertexes" to "vertices", except for those where the same message or comment already said "vertices" and it was thus inconsistent. I'll put the rest of the "vertices" changes in a separate PR to accept or reject as you like.

**Comment by alemuntoni** [comment]
Great, thank you again @rpavlik for this pull request!

**Pull request merged**